### PR TITLE
Mk master strip keywords terms

### PR DESCRIPTION
**Pull request by nimaai**
@eins78 please review, especially the migration

### CODE REVIEW

File: db/migrate/317_strip_keywords_terms.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            ::MigrationMetaDatumKeyword.where(keyword_id: keyword.id).each do |md|
+              # make a copy of the meta_datum
+              new_md = md.dup
+              # destroy the meta_datum for the stripped keyword
+              md.destroy
+              # point keyword_id for the new meta_datum to the other keyword
+              new_md.keyword_id = other_keyword.id
+              # create new meta_datum_keyword only if there is not one already for the same keyword and meta_datum
+              unless ::MigrationMetaDatumKeyword.find_by(keyword_id: new_md.keyword_id, meta_datum_id: new_md.meta_datum_id)
+                new_md.save
```
Comment (eins78) [inline db/migrate/317_strip_keywords_terms.rb]: please make sure that the timestamps stay the same… (maybe we even have to "trick" PG, or AR can help…)

File: app/models/keyword.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def to_s
   end
 
   before_save do
+    self.term = self.term.gsub(Madek::Constants::TRIM_WHITESPACE_REGEXP, '')
```
Comment (michalpodlecki) [inline app/models/keyword.rb]: `self` is necessary only for assigning a value, getting a value with `self` is redundant ;)

File: db/migrate/317_strip_keywords_terms.rb (hunk 1 above)
Comment (nimaai) [inline db/migrate/317_strip_keywords_terms.rb]: PG updates the timestamps via triggers. As I am disabling them in the migration alltogether, the perserverence of timestamps should be ensured. Nevertheless, I added some checks in the migration for that matter. See: https://github.com/Madek/madek-datalayer/pull/3/commits/d116f8a0493cd5fc19a34c83553e27186b957d59

File: app/models/keyword.rb (hunk 2 above)
Comment (nimaai) [inline app/models/keyword.rb]: Unless there is some local variable with the same name overshadowing the reader method which is not the case here, so you are right. ;-)

**Pull request closed without merging**